### DISCUSSION TITLE
validate BaseModel post init, post parse

### DISCUSSION THREAD

**Discussion by robbuckley** [pydantic/pydantic#3402] [Feature Request]
Hi, 
I've got a a use case where I'd like to run effectively a `root_validator` but on the Model post-initialisation. The reason for that is I need to use some model functionality in the validator, which isnt easy to replicate on the dict of fields i get with `root_validator`. Is there any way to do this? 

I can approximate it by overriding `__init__()` as suggested here https://github.com/samuelcolvin/pydantic/issues/691#issuecomment-515409301 but that has a number of drawbacks, mainly that its not possible to raise a  standard`ValidationError`

What would be great is if there would be another type of `root_validator` that gets called on the model, post initialisation,  (and after all "normal" `root_validators` have run), which gets passed the initialised instance, eg:

```
class MyModel(BaseModel):
  field1: int
  field2: str

  @root_validator(post_init=True)
  def check_model(cls, model: MyModel):
     ...
```

**Comment by uriyyo** [comment]
You can use `pre=True` and `pre=False` `root_validator`:
```py
from pydantic import BaseModel, root_validator, validator


class Model(BaseModel):
    a: int

    @root_validator(pre=True)
    def __before__(cls, values):
        print("before")
        return values

    @validator("a")
    def __parsing__(cls, v):
        print("parsing")
        return v

    @root_validator(pre=False, skip_on_failure=True)  # use skip_on_failure to call only when all fields are valid
    def __after__(cls, values):
        print("after")
        return values


print(Model(a=10))
```
```py
before
parsing
after
```
